coordinator/src/main/scala/filodb.coordinator/IngestionActor.scala

```diff
@@ -14,10 +14,11 @@ import kamon.Kamon
 import kamon.metric.MeasurementUnit
```

Comment:
Continue discussion: https://github.com/filodb/FiloDB/pull/1982#pullrequestreview-2730457989

Reply:
Thanks again for the offline discussion @sandeep6189.

tl;dr We will keep the recursive implementation because `Await`ing inside a `Monix` construct is a risky anti-pattern that will sometimes cause deadlock.